tensorflow-framework/src/test/java/org/tensorflow/framework/utils/ND.java

```diff
@@ -647,6 +731,97 @@ public static FloatNdArray sum(FloatNdArray a, Integer[] axes, boolean keepDims)
     }
   }
 
+  /**
+   * Sum all elements of an array
+   *
+   * @param a the array
+   * @return an a array with one element containing the sum.
+   */
+  public static DoubleNdArray sum(DoubleNdArray a) {
+    AtomicReference<Double> sum = new AtomicReference<>(0.);
+    a.scalars().forEach(f -> sum.set(sum.get() + f.getDouble()));
+    return NdArrays.scalarOf(sum.get());
+  }
+
+  /**
+   * Sum all elements of an array based on the specified axis
+   *
+   * @param a the array
+   * @param axis the axis to sum
+   * @return an a array the sum over the axis less the diemsnion
+   */
+  public static DoubleNdArray sum(DoubleNdArray a, int axis) {
+    return sum(a, axis, false);
+  }
+
+  /**
+   * Sum all elements of an array based on the specified axis
+   *
+   * @param a the array
+   * @param axis the axis to sum
+   * @param keepDims indicates whether the dimensions over the sum should be kept or not.
+   * @return an a array the sum over the axis
+   */
+  public static DoubleNdArray sum(DoubleNdArray a, int axis, boolean keepDims) {
+    Shape shape = a.shape();
+    int nDims = shape.numDimensions();
```

Comment:
Are we assuming this can't be `-1` in a `DoubleNdArray`? Seems reasonable, but I don't see it documented on `DoubleNdArray` or `NdArray`.

Reply:
👍 Perhaps ignore this issue in test code.